lib/licensed/sources/gradle.rb

```diff
@@ -122,61 +76,116 @@ def configurations
+        ensure
+          FileUtils.rm_rf(gradle_licenses_dir)
         end
       end
 
-      def self.gradle_command(*args, path:, executable:, configurations:)
-        gradle_build_file = File.read("build.gradle")
+      # Returns the cached url for the given dependency
+      def url_for(dependency)
+        @csv[csv_key(name: dependency.name, version: dependency.version)]
+      end
 
-        if !gradle_build_file.include? "dependency-license-report"
-          gradle_build_file = Licensed::Sources::Gradle.add_gradle_license_report_plugins_block(gradle_build_file)
+      # The Gradle::Runner class is a wrapper which provides
+      # an interface to run gradle commands with the init script initialized
+      class Runner
+        def initialize(root_path, configurations)
+          @root_path = root_path
+          @configurations = configurations
+          @executable = executable
         end
 
-        Dir.chdir(path) do
-          Tempfile.create(["license-", ".gradle"], path) do |f|
-            f.write(gradle_build_file)
-            f.write gradle_file(configurations)
-            f.close
-            Licensed::Shell.execute(executable, "-q", "-b", f.path, *args)
+        def run(*args)
+          Dir.chdir(@root_path) do
+            Tempfile.create(["init", ".gradle"], @root_path) do |f|
+              f.write(init_script(@configurations))
+              f.close
+              Licensed::Shell.execute(executable, "-q", "--no-configuration-cache", "--init-script", f.path, *args)
+            end
           end
         end
-      end
 
-      def self.gradle_file(configurations)
-        <<~EOF
+        def enabled?
+          !executable.to_s.empty?
+        end
 
-          import com.github.jk1.license.render.CsvReportRenderer
-          import com.github.jk1.license.filter.LicenseBundleNormalizer
+        private
 
-          final configs = #{configurations.inspect}
+        def executable
+          return @executable if defined?(@executable)
 
-          licenseReport {
-              configurations = configs
-              outputDir = "$projectDir/#{GRADLE_LICENSES_PATH}"
-              renderers = [new CsvReportRenderer()]
-              filters = [new LicenseBundleNormalizer()]
-          }
+          @executable = begin
+            gradlew = File.join(@root_path, "gradlew")
+            return gradlew if File.executable?(gradlew)
 
-          task printDependencies {
-              doLast {
-                  def dependencies = []
-                  configs.each {
-                      configurations[it].resolvedConfiguration.resolvedArtifacts.each { artifact ->
-                          def id = artifact.moduleVersion.id
-                          dependencies << "  { \\"group\\": \\"${id.group}\\", \\"name\\": \\"${id.name}\\", \\"version\\": \\"${id.version}\\" }"
+            "gradle" if Licensed::Shell.tool_available?("gradle")
+          end
+        end
+
+        def init_script(configurations)
```

Comment:
> it looks like you're getting rid of the add_grade_license_report_plugins_block function, as well as changing the name and contents of the created file.
> Would you mind adding some context behind the purpose of the changes and any specifics about the implementation?

Of course!

Here are the two reasons why I think an [init script](https://docs.gradle.org/current/userguide/init_scripts.html) is a great approach.

1. [Previously](https://github.com/github/licensed/pull/583/files#diff-155ac2e2d3b0e4ce3b140ae23797b529bba82dd79608e6b2c6134b25a11d8293L131), the `plugins` block was added at the beginning of the `build.gradle`. This approach raises an error when the `build.gradle` file has a `buildScript` block. 

    ```buildscript {} blocks must appear before any plugins {} blocks```

     By using an init script we can initialize the [Gradle License Plugin](https://github.com/jk1/Gradle-License-Report) for all subprojects without being prone to errors related to order of configuration keys in `build.gradle` files.

2. Previously, the modification of the build.gradle file was only done to one file, the one present in the folder where we run `licensed`.  This meant that the plugin and the `printDependencies` task was not initialized for subprojects. Also running `licensed` directly from a subproject folder was raising an error since no `settings.gradle` file was present. The `settings.gradle` file is usually present at the root of the build projet. 

    By using an init script we can initialize the `printDependencies` in all subprojects and safely execute it from to root by[ prefixing the subproject name](https://docs.gradle.org/current/userguide/intro_multi_project_builds.html#sec:executing_tasks_by_name). Ex: `subproject:printDependencies`